src/transforms/add_fields.rs

```diff
@@ -117,17 +116,21 @@ impl Transform for AddFields {
                 TemplateOrValue::Value(v) => v,
             };
             if self.overwrite {
-                if event.as_mut_log().insert(&key_string, value).is_some() {
+                if event
+                    .as_mut_log()
+                    .insert_path(key.clone().into(), value)
```

Comment:
Gosh here you are doing the next step of https://github.com/timberio/vector/blob/master/rfcs/2020-05-25-2692-more-usable-logevents.md :)